db/migrations/20190711091453_create_flip_table.sql

```diff
@@ -122,37 +120,42 @@ $$
 COMMENT ON FUNCTION get_latest_flip_bid_tab
     IS E'@omit';
 
+CREATE FUNCTION flip_bid_time_created(address_id INTEGER, bid_id NUMERIC) RETURNS TIMESTAMP AS
+$$
+SELECT api.epoch_to_datetime(MIN(block_timestamp))
+FROM public.headers
+         LEFT JOIN maker.flip_kick ON flip_kick.header_id = headers.id
+WHERE flip_kick.address_id = flip_bid_time_created.address_id
+  AND flip_kick.bid_id = flip_bid_time_created.bid_id
+$$
+    LANGUAGE sql;
+
+COMMENT ON FUNCTION flip_bid_time_created
+    IS E'@omit';
+
 
 -- +goose StatementBegin
 CREATE OR REPLACE FUNCTION maker.insert_updated_flip_guy() RETURNS TRIGGER AS
 $$
 BEGIN
-    WITH created AS (
-        SELECT created
-        FROM maker.flip
-        WHERE flip.bid_id = NEW.bid_id
-        ORDER BY flip.block_number
-        LIMIT 1
-    ),
-         diff_block AS (
-             select block_number, hash, block_timestamp
-             FROM public.headers
-             WHERE id = NEW.header_id
-         )
+    WITH diff_block AS (
+        SELECT block_number, block_timestamp
+        FROM public.headers
+        WHERE id = NEW.header_id
+    )
     INSERT
-    INTO maker.flip(bid_id, address_id, block_number, block_hash, guy, tic, "end", lot, bid, gal, tab, updated,
-                    created)
-    VALUES (NEW.bid_id, NEW.address_id, (SELECT block_number FROM diff_block), (SELECT hash FROM diff_block), NEW.guy,
-            (SELECT get_latest_flip_bid_tic(NEW.bid_id)),
-            (SELECT get_latest_flip_bid_end(NEW.bid_id)),
-            (SELECT get_latest_flip_bid_lot(NEW.bid_id)),
-            (SELECT get_latest_flip_bid_bid(NEW.bid_id)),
-            (SELECT get_latest_flip_bid_gal(NEW.bid_id)),
-            (SELECT get_latest_flip_bid_tab(NEW.bid_id)),
+    INTO maker.flip (bid_id, address_id, block_number, guy, tic, "end", lot, bid, gal, tab, updated, created)
+    VALUES (NEW.bid_id, NEW.address_id, (SELECT block_number FROM diff_block), NEW.guy,
+            get_latest_flip_bid_tic(NEW.bid_id),
```

Comment:
Maybe I should have done some of this reformatting/refactoring in a separate commit 😅 but the only behavior change in all these `insert_updated_x` trigger functions are cascading changes from updating the `flip` table. I.e. removing `block_hash` and updating the PK.